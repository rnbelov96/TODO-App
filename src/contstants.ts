import { FullStateType } from './types/general-types';

const enum FilterStatus {
  ALL = 'ALL',
  ACTIVE = 'ACTIVE',
  COMPLETED = 'COMPLETED',
}

export const mockFullInitiateState: FullStateType = {
  app: {
    todos: [
      {
        text: 'Go for a walk',
        done: false,
        id: '6436c8b9-58c6-4e94-92f2-e4fac524a729',
      },
      {
        text: 'Read the book',
        done: false,
        id: '6436c8b9-58c6-4e94-92f2-e4fac524a730',
      },
      {
        text: 'Call my friends',
        done: true,
        id: '6436c8b9-58c6-4e94-92f2-e4fac524a731',
      },
    ],
    inputText: '',
    filterBy: FilterStatus.COMPLETED,
  },
};

export default FilterStatus;
