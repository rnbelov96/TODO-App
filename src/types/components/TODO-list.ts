import { TODOType } from '../general-types';

export type TODOListPropsType = {
  todosToShow: TODOType[];
  onChange: (id: string) => void;
  onDelete: (id: string) => void;
};
