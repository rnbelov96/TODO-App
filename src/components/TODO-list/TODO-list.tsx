import { AppActionCreators } from '@/redux/app';
import { getTODOSToShow } from '@/redux/app/selectors';
import { TODOListPropsType } from '@/types/components/TODO-list';
import { FullStateType } from '@/types/general-types';
import { AppActionType } from '@/types/redux/app-reducer';
import React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

export const PureTODOList: React.FunctionComponent<TODOListPropsType> = ({
  todosToShow,
  onChange,
  onDelete,
}) => (
  <>
    {todosToShow.map(el => (
      <div
        className={el.done ? 'todoItem todoItem_done' : 'todoItem'}
        key={el.id}
      >
        <input
          checked={el.done}
          onChange={() => onChange(el.id)}
          type="checkbox"
        />
        {el.text}
        <button
          type="button"
          onClick={() => onDelete(el.id)}
          className="todoItem__deleteButton"
        >
          x
        </button>
      </div>
    ))}
  </>
);

function mapStateToProps(state: FullStateType) {
  return {
    todosToShow: getTODOSToShow(state),
  };
}

function mapDispatchToProps(dispatch: Dispatch<AppActionType>) {
  return {
    onChange: (id: string) => dispatch(AppActionCreators.changeTodoStatus(id)),
    onDelete: (id: string) => dispatch(AppActionCreators.deleteTODO(id)),
  };
}

export const TODOList = connect(mapStateToProps, mapDispatchToProps)(PureTODOList);
