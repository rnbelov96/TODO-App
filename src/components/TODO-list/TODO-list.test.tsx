import { mockFullInitiateState } from '@/contstants';
import React from 'react';
import renderer from 'react-test-renderer';
import { PureTODOList } from './TODO-list';

describe('TODOList', () => {
  test('should render todo list', () => {
    const tree = renderer
      .create(
        <PureTODOList
          todosToShow={mockFullInitiateState.app.todos}
          onChange={() => {}}
          onDelete={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
