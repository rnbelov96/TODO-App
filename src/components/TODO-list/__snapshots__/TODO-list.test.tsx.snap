// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TODOList should render todo list 1`] = `
Array [
  <div
    className="todoItem"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Go for a walk
    <button
      className="todoItem__deleteButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>,
  <div
    className="todoItem"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Read the book
    <button
      className="todoItem__deleteButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>,
  <div
    className="todoItem todoItem_done"
  >
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Call my friends
    <button
      className="todoItem__deleteButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>,
]
`;
